src/main/java/org/elasticsearch/common/joda/Joda.java

```diff
@@ -133,6 +134,10 @@ public static FormatDateTimeFormatter forPattern(String input, Locale locale) {
             formatter = ISODateTimeFormat.yearMonth();
         } else if ("yearMonthDay".equals(input) || "year_month_day".equals(input)) {
             formatter = ISODateTimeFormat.yearMonthDay();
+        } else if ("epochSecond".equals(input) || "epoch_second".equals(input)) {
```

Comment:
In 2.0 ParseField no longer supports camelCase IIRC. The advantage with using ParseField is that it standardises the checks that we do to make it consistent across the codebase (or will once all the Parsers have been changed to use it). It also makes deprecating an old option name in favour of a new one, much easier as we don't end up with these messy if statements with multiple ||'s. And also means its much easier to correlate the builder classes with the Parser since they can both reference a ParseField constant and the builder will alway use the preferred name for the option.